vendor/plugins/edu.harvard.dce.paella.heartbeatSender/heartbeat_sender.js

```diff
@@ -0,0 +1,68 @@
+// A version of the deprecated es.upv.paella.UserTrackingCollectorPlugIn, 
+// shaved down to just send the heartbeat.
+
+var classDef = {
+  heartbeatTimer:null,
+
+  getName: function() {
+    return "edu.harvard.dce.paella.heartbeatSender";
+  },
+
+  setup: function() {
+    var thisClass = this;
+  
+    if (this.config.heartBeatTime > 0) {
+      this.heartbeatTimer = new base.Timer(
+        registerHeartbeat, this.config.heartBeatTime
+      );
+      this.heartbeatTimer.repeat = true;
+    }
+
+    function registerHeartbeat(timer) {
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', getHeartbeatURL());
+      xhr.send();
+    }
+
+    function getHeartbeatURL() {
+      var videoCurrentTime = parseInt(
+        paella.player.videoContainer.currentTime() +
+        paella.player.videoContainer.trimStart(),
+        10
+      );
+
+      var url = 'https://';
+      url += location.host + '/';
+      url += 'usertracking/?';
+      url += queryStringFromDict({
+        _method: 'PUT',
+        id: paella.player.videoIdentifier,
+        type: 'HEARTBEAT',
+        in: videoCurrentTime,
```

Comment:
Putting a single quote around the "in" key above prevents javascript from complaining about using the reserved word.